### PR TITLE
Node - Week13 Hometask

### PR DESCRIPTION
**Pull request by AnnaSolovykh**

### CODE REVIEW

File: controllers/sessionController.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+const User = require("../models/User");
+const parseVErr = require("../utils/parseValidationErrs");
+const csrf = require("host-csrf");
+
+const registerShow = (req, res) => {
+  res.render("register");
+};
+
+const registerDo = async (req, res, next) => {
```
Comment (olearyma) [inline controllers/sessionController.js:9]: I encountered some validation issues but it turns out the provided code in the lesson had some errors.  

For more information see  #ferret-announcements in Slack
This updated code should fix the issues.
```
const registerDo = async (req, res, next) => {
  let validation_errors = false
  if (req.body.password != req.body.password1) {
    req.flash("error", "The passwords entered do not match.");
    validation_errors = true
  }

  try {
    await User.create(req.body);
  } catch (e) {
    if (e.constructor.name === "ValidationError") {
      parse_v(e, req);
    } else if (e.name === "MongoServerError" && e.code === 11000) {
      req.flash("error", "That email address is already registered.");
    } else {
      return next(e);
    }
    validation_errors = true
  }
  
  // if there are no validation errors then redirect to main paige 
  // else pass errors to be rendered in the register view

  if (! validation_errors)
  {
    res.redirect("/");
  }
  else
  {
    return res.render("register", {errors: req.flash("error")})
  }


};
```

Comment (AnnaSolovykh) [reply controllers/sessionController.js:9]: Thanks you!

**Review by olearyma — APPROVED** [review]
Nice work implementing passport into the project

**Comment by AnnaSolovykh** [comment]
> Nice work implementing passport into the project

Thank you!

**Pull request merged**